Technical-Steering-Committee/best-practices/build-and-test.md

```diff
@@ -0,0 +1,17 @@
+# Build and Test
+
+## Build
+If a Zowe component or a repository requires source code compilation, this repository should contain automated build system able to compile the software from source, automatically handle dependencies and produce installation packages. The build tools in use should be only FLOSS, free to acquire and use for any purposes. It is recommended to select tools already popular within Zowe, such as Gradle, npm, GitHub Actions, etc. For specific needs, it is possible to select alternatives, given they provide unique and necessary advantages.
+
+## Test
+Every Zowe component or repository should contain an automated test suit that is executed on every change or code build. Ideally the automated test suit should consist of the following categories of tests - unit tests, integration tests and system tests. The tests should cover both functional, i.e. validating a feature functionality, as well as non-functional, i.e. performance, reliability, resilience, etc. 
+
+## Code Analysis
+It is recommended, whenever reasonable, each repository to make use of static code analysis and vulnerability scans per tools availability and requirement. For repositories that contain mainly Java and/or Javascript code, for example, the recommendation is to enable Sonarcloud scans for each change. All reported issues should be addressed either by resolving them or by indicating an exception justification.
```

Comment:
We are gradually moving to the requirement to do so. Actually, we already agreed that each squad needs to select and use at least one of such tools. We recommend t depend either on SonarCloud or CodeQL. One of the benefits of codeQL is that it is relatively easy to create and share new rules specific to our understanding of Zowe problems. 

As for the exception justification, I believe we will need to work with Linux Foundation in order to gain access to administrator interface in SonarCloud to keep the information about exceptions there.